File: Enums/InputTypeEstr.cs

namespace DbAdm.Enums
{
    /// <summary>
    /// input type, mapping to Code.type=InputType
    /// </summary>
    public class InputTypeEstr
    {
        //normal text input, no validation
        public const string Hide = "H";
        public const string Text = "T";
        public const string TextArea = "TA";
        public const string Numeric = "N";
        public const string Select = "S";
        public const string CheckBox = "C";
        public const string Radio = "R";
        public const string Date = "D";
        public const string DateTime = "DT";
        public const string File = "F";
        public const string Sort = "SO";
        public const string Modal = "MO";
        public const string Password = "PWD";
        public const string ReadOnly = "RO";
    }
}

File: _ts/base/enums/InputTypeEstr.ts
//輸入欄位種類, 對應 Base InputTypeEstr 
export default class InputTypeEstr {
    // 原始物件字面量中的常數值被轉換為 static readonly 屬性，以符合 class 語法
    // 且保持 InputTypeEstr.Check 這樣的外部存取方式。

    static readonly Check: string = "check";
    static readonly Date: string = "date";
    static readonly DateTime: string = "dt";
    static readonly Decimal: string = "dec";
    static readonly File: string = "file";
    static readonly Hide: string = "hide";
    static readonly Html: string = "html";
    static readonly Integer: string = "int";
    static readonly Link: string = "link";
    static readonly Modal: string = "modal";
    static readonly Password: string = "pwd";
    static readonly Radio: string = "radio";
    static readonly Read: string = "read";
    static readonly Select: string = "select";
    static readonly Sort: string = "sort";
    static readonly Text: string = "text";
    static readonly Textarea: string = "textarea";

}